README.md

```diff
@@ -1,2 +1,207 @@
+   Defaults to a new tmp directory.
+ - **buildDir**: The directory to build the application in. Defaults to
+   `${workingDir}/build`
+ - **repoDir**: The directory for a flatpak repo, can be used to publish to an
+   existing repo. Defaults to `${workingDir}/repo`
+ - **cleanTmpdirs**: Cleanup any tmp directories created during the build on
+   process exit. Defaults to true. Set false for easier debugging.
+ - **autoInstallRuntime**: Install/update the runtime while building. Defaults
+   to true if runtimeFlatpakref is set in the manifest.
+ - **autoInstallSdk**: Install/update the sdk while building. Defaults
+   to true if sdkFlatpakref is set in the manifest.
+ - **autoInstallBase**: Install/update the base app while building. Defaults
+   to true if baseFlatpakref is set in the manifest.
+ - **gpgSign**: The gpg key to use to sign the flatpak repo and bundle file.
+ - **gpgHomedir**: The gpg homedir to use when signing.
+ - **subject**: The single line subject to use for the flatpak repo commit
+   message.
+ - **body**: The description to use for the flatpak repo commit message.
+ - **bundleRepoUrl**: Repo url for the single file bundle. Installing the bundle
+   will automatically configure a remote for this URL.
+ - **extraFlatpakBuilderArgs**: List of extra arguments to pass to the
+   [flatpak-builder](http://flatpak.org/flatpak/flatpak-docs.html#flatpak-builder) command.
+ - **extraFlatpakBuildExportArgs**: List of extra arguments to pass to the
+   [flatpak build-export](http://flatpak.org/flatpak/flatpak-docs.html#flatpak-build-export) command.
+ - **extraFlatpakBuildBundleArgs**: List of extra arguments to pass to the
+   [flatpak build-bundle](http://flatpak.org/flatpak/flatpak-docs.html#flatpak-build-bundle) command.
+
+### Logging
+To turn on debugging output set the DEBUG environment variable
+```
+DEBUG=flatpak-bundler npm run my-flatpak-command
+```
+
+## Examples
+
+#### Hello world
+
+```js
+// Write a hello world script to disk
+const fs = require('fs')
+fs.writeFileSync('hello',
+`#!/bin/bash
+echo "Hello, world!"`, { mode: 0o755 })
+
+// Make a flapak with it!
+const flatpakBundler = require('flatpak-bundler')
+flatpakBundler.bundle({
+  id: 'org.world.Hello',
+  runtime: 'org.freedesktop.Platform',
+  runtimeVersion: '1.4',
+  runtimeFlatpakref: 'https://raw.githubusercontent.com/mattdangerw/flatpak-bundler/master/refs/freedesktop-runtime-1.4.flatpakref',
+  sdk: 'org.freedesktop.Sdk',
+  files: [
+    ['hello', '/bin/hello']
+  ]
+}, {
+  bundlePath: 'hello.flatpak'
+}, function (error) {
+  if (error) {
+    console.error('Error building flatpak', error)
+    return
+  }
+  console.log('Flatpak built successfully')
+})
+```
+
+#### Electron app
+
+```js
+const flatpakBundler = require('flatpak-bundler')
+
+flatpakBundler.bundle({ // Manifest
+  id: 'org.world.Hello',
+  base: 'io.atom.electron.BaseApp', // Electron base application
+  baseFlatpakref: FIXME, // So we can auto install the runtime
```

Comment:
Not sure. There's the `autoInstallBase` build option that controls this, which I default to true if `baseFlatpakref` is set in your manifest cause I figure that's what a user would generally want.

But the goal with this split is you could use the same app manifest (where you want to pull your runtimes from is unlikely to change) with different types of builds (say `autoInstallBase: false` to just use your local hacked up version).

But maybe where you want to pull your platform/sdk/base from is really more of a build option? And I should just remove these from the manifest entirely?

Another thing I was unsure of was whether to decouple the auto install and auto update options. I wanted to avoid an explosion of properties, but maybe there's some reason you'd want one but not the other.

The only thing I think is worth protecting here is avoiding any setup. A user should be able to go to any machine as long as node and flatpak are available, build their electron app as a flatpak without any further manual installation of dependencies.